packages/frontend/src/pages/JoinOrganization.tsx

```diff
@@ -0,0 +1,130 @@
+import { Helmet } from 'react-helmet';
+import { useHistory } from 'react-router-dom';
+import { BigButton } from '../components/common/BigButton';
+import Page from '../components/common/Page/Page';
+import PageSpinner from '../components/PageSpinner';
+import { useOrganisationCreateMutation } from '../hooks/organisation/useOrganisationCreateMutation';
+import useAllowedOrganizations from '../hooks/user/useAllowedOrganizations';
+import { useJoinOrganizationMutation } from '../hooks/user/useJoinOrganizationMutation';
+import { useApp } from '../providers/AppProvider';
+import LightdashLogo from '../svgs/lightdash-black.svg';
+import {
+    CardWrapper,
+    FormWrapper,
+    Logo,
+    LogoWrapper,
+    Subtitle,
+    Title,
+} from './SignUp.styles';
+
+export const JoinOrganizationPage: FC = () => {
+    const { health, user } = useApp();
+    const history = useHistory();
+    const { isLoading: isLoadingAllowedOrgs, data: allowedOrgs } =
+        useAllowedOrganizations();
+    const {
+        mutate: createOrg,
+        isLoading: isCreatingOrg,
+        isSuccess: hasCreatedOrg,
+    } = useOrganisationCreateMutation();
+    const {
+        mutate: joinOrg,
+        isLoading: isJoiningOrg,
+        isSuccess: hasJoinedOrg,
+    } = useJoinOrganizationMutation();
+
+    useEffect(() => {
+        const isPR = health.data?.mode === LightdashMode.PR;
+        const isAllowedToJoinOrgs = allowedOrgs && allowedOrgs.length > 0;
+        const userHasOrg = user.data && !!user.data.organizationUuid;
+        if (
+            !isCreatingOrg &&
+            !isLoadingAllowedOrgs &&
+            !userHasOrg &&
+            (isPR || !isAllowedToJoinOrgs)
+        ) {
+            createOrg({ name: '' });
+        }
+    }, [
+        health,
+        allowedOrgs,
+        createOrg,
+        isCreatingOrg,
+        user,
+        isLoadingAllowedOrgs,
+    ]);
+
+    useEffect(() => {
+        if (hasCreatedOrg || hasJoinedOrg) {
+            history.push('/');
+        }
+    }, [hasCreatedOrg, hasJoinedOrg]);
+
+    if (health.isLoading || isLoadingAllowedOrgs || isCreatingOrg) {
+        return <PageSpinner />;
+    }
+
+    const disabled = isCreatingOrg || isJoiningOrg;
+
+    return (
+        <Page isFullHeight>
+            <Helmet>
+                <title>Join a workspace - Lightdash</title>
+            </Helmet>
+            <LogoWrapper>
+                <Logo src={LightdashLogo} alt="lightdash logo" />
+            </LogoWrapper>
+            <FormWrapper>
+                <CardWrapper elevation={2}>
+                    <Title>Join a workspace</Title>
+                    <Subtitle>
+                        The workspaces below are open to anyone with a{' '}
+                        <b>@{(user.data?.email || '').split('@')[1]}</b> domain.
+                        Select a workspace to join, or create a new one!
+                    </Subtitle>
+                    {allowedOrgs?.map((org) => (
+                        <Card
```

Comment:
Is there a Mantine component that replaces Blueprint `Card`?
I'm keeping the inline styling since we are moving away from styled components.